### PR TITLE
Add spotbug checks for pulsar-client-api module

### PR DESCRIPTION
**Pull request by BewareMyPower**
### Motivation

Add spotbugs plugin for pulsar-client-api module.

### Modifications

The most potential bugs are related to the raw array's getter/setter/constructor and the rest one is just a default method of an interface, so we just ignore these spotbugs.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/EncryptionKeyInfo.java [hunk 1]
```diff
@@ -39,16 +39,16 @@ public EncryptionKeyInfo() {
     }
 
     public EncryptionKeyInfo(byte[] key, Map<String, String> metadata) {
-        this.key = key;
+        this.key = key.clone();
```
Comment (eolivelli) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/EncryptionKeyInfo.java:42]: IMO it is better to add a suppresswarnings here and do not clone the array.
Even if we are not in an hot path

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/EncryptionKeyInfo.java (hunk 1 above)
Comment (BewareMyPower) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/EncryptionKeyInfo.java:42]: OK, I'll ignore the setter/getter of the array.

**Review by eolivelli — APPROVED** [review]
Lgtm

**Review by codelipenghui — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**